NewsPassID Bid Adapter: refactor

omlid opened the pull request:
<!--
Thank you for your pull request! 

Please title your pull request like this: 'Module: Change', eg 'Fraggles Bid Adapter: support fragglerock'

Please make sure this PR is scoped to one change or you may be asked to resubmit. 
 
Please make sure any added or changed code includes tests with greater than 80% code coverage. 

See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.

For any user facing change, submit a link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: also submit your bidder parameter documentation as noted in https://docs.prebid.org/dev-docs/bidder-adaptor.html#submitting-your-adapter -->
- [x] Updated bidder adapter  <!--  IMPORTANT: (1) consider whether you need to upgrade your bidder parameter documentation in https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders and (2) if you have a Prebid Server adapter, please consider whether that should be updated as well. --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
<!-- Describe the change proposed in this pull request -->
Refactor of the newspassidBidAdapter. Previous provider Ozone is no longer an underlying tech provider for the LMC NewsPassID initiative.

<!-- For new bidder adapters, please provide the following
- contact email of the adapter’s maintainer
- test parameters for validating bids:
```
{
  bidder: 'newspassid',
  params: {
    accountId: '123456',
    groupId: 'test-group1'
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](https://github.com/prebid/Prebid.js/blob/master/integrationExamples/gpt/hello_world.html) sample page. -->


## Other information
<!-- References to related PR or issue #s, @mentions of the person or team responsible for reviewing changes, etc. -->
[Link to Documentation PR](https://github.com/prebid/prebid.github.io/pull/5962)

ChrisHuie commented:
@omlid I don't understand this pr? You are removing your adapter files and changing the markdown file name?

patmmccann commented on modules/newspassidBidAdapter.js:
```diff
@@ -1,674 +1,164 @@
+const DEFAULT_NET_REVENUE = true;
+const DEFAULT_TTL = 300;
+const ENDPOINT_URL = 'https://npid.amspbs.com/v0/bid/request';
+const GVL_ID = 1317;
```
this is aditude, and most of these changes are breaking, do you want to target the ten branch?

omlid replied:
@patmmccann Without knowledge on v10 timeline, I need this in v9. Previous provider to the NewsPassID initiative was Ozone. That relationship has been dead for >= 6 months. The current `newspassid` bidder in master fails to respond to bid requests for any publisher. We're racing here to fix.

A change in parameters should gracefully catch via `isBidRequestValid` and not break the golden path anyways, right? If anything, we're reducing wasteful network requests for those with this version of the bid adapter still live.

patmmccann replied:
instead of requiring new configuration, why don't you just take the old required params 
````
| `publisherId`    | required | The publisher ID.  | `"4204204201"` | `string` |
| `placementId`    | required | The placement ID.  | `"0420420421"` | `string` |

```
and treat them as aliases for the new required params? 

````
| `publisherId`    | required | The publisher ID.  | `"4204204201"` | `string` |
| `placementId`    | required | The placement ID.  | `"0420420421"` | `string` |

````

Pubs shouldnt be required to update configuration in minor versions

patmmccann commented on modules/newspassidBidAdapter.js:
```diff
@@ -1,674 +1,164 @@
+    const imp = buildImp(bidRequest, context);
+    deepSetValue(imp, 'ext.newspassid', {
+      accountId: resolveNewpassidAccountId(bidRequest),
+      groupId: bidRequest.params.groupId,
```
just accept the ozone placement id here.

omlid replied:
Updated. Thank you.

omlid commented on modules/newspassidBidAdapter.js:
```diff
@@ -1,674 +1,164 @@
+ * @returns {string|null}
+ */
+const getGlobalAccountIdOrNull = () => {
+  const globalAccountId = config.getConfig('newspassid.accountId');
```
Bid adapter code is updated. We'll have backend sorted out today. Will report back in this thread.

omlid commented:
@patmmccann Ready for re-review.

Updates:
- Moved to `placementId` and `publisherId` to address concerns
- We're accepting `imp[..].ext.newspassid.publisher` on the backend now ("publisher" is intentional)
- Updated tests, added a few more to check for prioritization of the publisherId value
- Updated newspassid documentation (in this wrapper and the prebid.github.io repo)
